taichi/program/state_flow_graph.cpp

```diff
@@ -790,14 +801,18 @@ std::unordered_set<int> StateFlowGraph::fuse_range(int begin, int end) {
     if (!fused[i]) {
       // Fuse no more than one task into task i
       bool i_updated = false;
+      int remaining_fuses = get_initial_remaining_fuses();
       for (auto &edge : nodes[i]->output_edges.get_all_edges()) {
```

Comment:
Note that "Iterators of nodes[i]->output_edges may be invalidated" upon `do_fuse`, so we may need to break this loop or rewrite the loop and handle the iterators carefully.

Reply:
Because of `replace_reference` in `do_fuse`, it's hard to handle the iterators here. Shall we loop over all edges for `remaining_fuses` times?

Reply:
Ah you are right, and I believe the latest decision is not to change this part?

Reply:
Yes, we decide to not change this part.